business/core/home/home.go

```diff
@@ -0,0 +1,203 @@
+package home
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ardanlabs/service/business/core/event"
+	"github.com/ardanlabs/service/business/core/user"
+	"github.com/ardanlabs/service/business/data/order"
+	"github.com/ardanlabs/service/business/data/transaction"
+	"github.com/ardanlabs/service/foundation/logger"
+	"github.com/google/uuid"
+)
+
+// Set of error variables for CRUD operations.
+var (
+	ErrNotFound    = errors.New("home not found")
+	ErrInvalidUser = errors.New("user not valid")
+)
+
+// Storer interface declares the behaviour this package needs to persist and
+// retrieve data
+type Storer interface {
+	ExecuteUnderTransaction(tx transaction.Transaction) (Storer, error)
+	Create(ctx context.Context, nh Home) error
+	Update(ctx context.Context, hme Home) error
+	Delete(ctx context.Context, hme Home) error
+	Query(ctx context.Context, filter QueryFilter, orderBy order.By, pageNumber int, rowsPerPage int) ([]Home, error)
+	Count(ctx context.Context, filter QueryFilter) (int, error)
+	QueryByID(ctx context.Context, homeID uuid.UUID) (Home, error)
+	QueryByUserID(ctx context.Context, userID uuid.UUID) ([]Home, error)
+}
+
+// UserCore interface declares the behavior this package needs from the user
+// core domain.
+type UserCore interface {
+	ExecuteUnderTransaction(tx transaction.Transaction) (*user.Core, error)
+	QueryByID(ctx context.Context, userID uuid.UUID) (user.User, error)
+}
+
+// =============================================================================
+
+// Core manages the set of APIs for home api access.
+type Core struct {
+	log     *logger.Logger
+	usrCore *user.Core
+	evnCore *event.Core
+	storer  Storer
+}
+
+func NewCore(log *logger.Logger, evnCore *event.Core, usrCore *user.Core, storer Storer) *Core {
+	core := Core{
+		log:     log,
+		usrCore: usrCore,
+		evnCore: evnCore,
+		storer:  storer,
+	}
+
+	return &core
+}
+
+// ExecuteUnderTransaction constructs a new Core value that will use the
+// specified transaction in any store related calls.
+func (c *Core) ExecuteUnderTransaction(tx transaction.Transaction) (*Core, error) {
+	storer, err := c.storer.ExecuteUnderTransaction(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	usrCore, err := c.usrCore.ExecuteUnderTransaction(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	c = &Core{
+		storer:  storer,
+		usrCore: usrCore,
+		evnCore: c.evnCore,
+		log:     c.log,
+	}
+
+	return c, nil
+}
+
+// Create adds a Home to the database. It returns the created Home with
+// fields like ID and DateCreated populated
+func (c *Core) Create(ctx context.Context, nh NewHome) (Home, error) {
```

Comment:
I hate this comment and I know you copied it. Plus you are missing a closing period. 

We probably don't want to say database since the store abstraction might not be a database.

I think the second sentence is about us returning a complete Home with it's unique id and the date created.

We should fix this in all domains.

Reply:
Will do then. Comments seem weird all over the place, I'll fix whatever I found odd.